:ALEFix (eslint) recently does nothing

I've been using ale for a while now, I'm not sure when but at some point recently `:ALEFix` seems to have stopped working.

To be clear `:ALELint` functionality still works, and I still get the errors shown in the gutter and in the status line.

There is no obvious error when running `:ALEFix` - just nothing happens.

I've tried updating ale just in case, I've also reinstalled eslint by using `npm install -g eslint`.

I've checked my config which reads:
```vimscript
let g:ale_fixers = {
\   'javascript': ['eslint'],
\}

let g:ale_linters = {
\   'javascript': ['eslint'],
\}
```
I've also check with `:echo ale_fixers` which outputs `{'javascript': ['eslint']}`.

Also I can see that eslint itself works as I can run it from the command line - or even better I can run from neovim: `:!eslint --fix %` and the fixes are run correctly.


```
 Current Filetype: javascript
Available Linters: ['eslint', 'flow', 'jscs', 'jshint', 'standard', 'xo']
  Enabled Linters: ['eslint']
 Linter Variables:

let g:ale_javascript_eslint_executable = 'eslint'
let g:ale_javascript_eslint_options = ''
let g:ale_javascript_eslint_suppress_eslintignore = 0
let g:ale_javascript_eslint_use_global = 0
 Global Variables:

let g:ale_echo_cursor = 1
let g:ale_echo_msg_error_str = 'Error'
let g:ale_echo_msg_format = '%s'
let g:ale_echo_msg_warning_str = 'Warning'
let g:ale_enabled = 1
let g:ale_fix_on_save = 0
let g:ale_fixers = {'javascript': ['eslint']}
let g:ale_keep_list_window_open = 0
let g:ale_lint_delay = 200
let g:ale_lint_on_enter = 1
let g:ale_lint_on_save = 1
let g:ale_lint_on_text_changed = 'always'
let g:ale_linter_aliases = {}
let g:ale_linters = {'javascript': ['eslint']}
let g:ale_open_list = 0
let g:ale_set_highlights = 1
let g:ale_set_loclist = 1
let g:ale_set_quickfix = 0
let g:ale_set_signs = 1
let g:ale_sign_column_always = 0
let g:ale_sign_error = '>>'
let g:ale_sign_offset = 1000000
let g:ale_sign_warning = '--'
let g:ale_statusline_format = ['%d error(s)', '%d warning(s)', 'OK']
let g:ale_warn_about_trailing_whitespace = 1
  Command History:

(executable check - success) eslint
(started) ['/bin/zsh', '-c', '''eslint'' -f unix --stdin --stdin-filename ''/Users/pgrove/git/ddi-guard-jira3/master/tests/active-directory.js'' < ''/var/folders/w_/nwg5jb2d5h19lhpv_rhx3c4c0000gn/T/nvimeHsmLZ/2/active-directory.js''']
(finished - exit code 1) ['/bin/zsh', '-c', '''eslint'' -f unix --stdin --stdin-filename ''/Users/pgrove/git/ddi-guard-jira3/master/tests/active-directory.js'' < ''/var/folders/w_/nwg5jb2d5h19lhpv_rhx3c4c0000gn/T/nvimeHsmLZ/3/active-directory.js''']

<<<OUTPUT STARTS>>>
/Users/pgrove/git/ddi-guard-jira3/master/tests/active-directory.js:2:1: Use the global form of 'use strict'. [Error/strict]
/Users/pgrove/git/ddi-guard-jira3/master/tests/active-directory.js:2:1: Unexpected var, use let or const instead. [Error/no-var]
/Users/pgrove/git/ddi-guard-jira3/master/tests/active-directory.js:4:1: Unexpected var, use let or const instead. [Error/no-var]
/Users/pgrove/git/ddi-guard-jira3/master/tests/active-directory.js:8:38: Unexpected function expression. [Error/prefer-arrow-callback]
/Users/pgrove/git/ddi-guard-jira3/master/tests/active-directory.js:9:22: Unexpected function expression. [Error/prefer-arrow-callback]
/Users/pgrove/git/ddi-guard-jira3/master/tests/active-directory.js:12:5: Expected object keys to be in ascending order. 'port' should be before 'target'. [Warning/sort-keys]
/Users/pgrove/git/ddi-guard-jira3/master/tests/active-directory.js:16:1: Trailing spaces not allowed. [Error/no-trailing-spaces]
/Users/pgrove/git/ddi-guard-jira3/master/tests/active-directory.js:17:27: Unexpected function expression. [Error/prefer-arrow-callback]
/Users/pgrove/git/ddi-guard-jira3/master/tests/active-directory.js:20:5: Expected object keys to be in ascending order. 'port' should be before 'target'. [Warning/sort-keys]
/Users/pgrove/git/ddi-guard-jira3/master/tests/active-directory.js:24:1: Trailing spaces not allowed. [Error/no-trailing-spaces]
/Users/pgrove/git/ddi-guard-jira3/master/tests/active-directory.js:25:26: Unexpected function expression. [Error/prefer-arrow-callback]
/Users/pgrove/git/ddi-guard-jira3/master/tests/active-directory.js:28:5: Expected object keys to be in ascending order. 'port' should be before 'target'. [Warning/sort-keys]
/Users/pgrove/git/ddi-guard-jira3/master/tests/active-directory.js:32:1: Trailing spaces not allowed. [Error/no-trailing-spaces]
/Users/pgrove/git/ddi-guard-jira3/master/tests/active-directory.js:33:31: Unexpected function expression. [Error/prefer-arrow-callback]
/Users/pgrove/git/ddi-guard-jira3/master/tests/active-directory.js:36:5: Expected object keys to be in ascending order. 'port' should be before 'target'. [Warning/sort-keys]
/Users/pgrove/git/ddi-guard-jira3/master/tests/active-directory.js:40:1: Trailing spaces not allowed. [Error/no-trailing-spaces]
/Users/pgrove/git/ddi-guard-jira3/master/tests/active-directory.js:41:28: Unexpected function expression. [Error/prefer-arrow-callback]
/Users/pgrove/git/ddi-guard-jira3/master/tests/active-directory.js:44:5: Expected object keys to be in ascending order. 'port' should be before 'target'. [Warning/sort-keys]
/Users/pgrove/git/ddi-guard-jira3/master/tests/active-directory.js:48:1: Trailing spaces not allowed. [Error/no-trailing-spaces]
/Users/pgrove/git/ddi-guard-jira3/master/tests/active-directory.js:49:32: Unexpected function expression. [Error/prefer-arrow-callback]
/Users/pgrove/git/ddi-guard-jira3/master/tests/active-directory.js:52:5: Expected object keys to be in ascending order. 'port' should be before 'target'. [Warning/sort-keys]
/Users/pgrove/git/ddi-guard-jira3/master/tests/active-directory.js:57:1: Trailing spaces not allowed. [Error/no-trailing-spaces]
/Users/pgrove/git/ddi-guard-jira3/master/tests/active-directory.js:58:28: Unexpected function expression. [Error/prefer-arrow-callback]
/Users/pgrove/git/ddi-guard-jira3/master/tests/active-directory.js:59:18: Unexpected function expression. [Error/prefer-arrow-callback]
/Users/pgrove/git/ddi-guard-jira3/master/tests/active-directory.js:62:5: Expected object keys to be in ascending order. 'dnsDomainName' should be before 'username'. [Warning/sort-keys]
/Users/pgrove/git/ddi-guard-jira3/master/tests/active-directory.js:66:1: Trailing spaces not allowed. [Error/no-trailing-spaces]
/Users/pgrove/git/ddi-guard-jira3/master/tests/active-directory.js:67:30: Unexpected function expression. [Error/prefer-arrow-callback]
/Users/pgrove/git/ddi-guard-jira3/master/tests/active-directory.js:70:5: Expected object keys to be in ascending order. 'dnsDomainName' should be before 'username'. [Warning/sort-keys]
/Users/pgrove/git/ddi-guard-jira3/master/tests/active-directory.js:74:1: Trailing spaces not allowed. [Error/no-trailing-spaces]
/Users/pgrove/git/ddi-guard-jira3/master/tests/active-directory.js:75:30: Unexpected function expression. [Error/prefer-arrow-callback]
/Users/pgrove/git/ddi-guard-jira3/master/tests/active-directory.js:78:5: Expected object keys to be in ascending order. 'dnsDomainName' should be before 'username'. [Warning/sort-keys]
/Users/pgrove/git/ddi-guard-jira3/master/tests/active-directory.js:82:1: Trailing spaces not allowed. [Error/no-trailing-spaces]
/Users/pgrove/git/ddi-guard-jira3/master/tests/active-directory.js:83:42: Unexpected function expression. [Error/prefer-arrow-callback]
/Users/pgrove/git/ddi-guard-jira3/master/tests/active-directory.js:86:5: Expected object keys to be in ascending order. 'dnsDomainName' should be before 'username'. [Warning/sort-keys]

34 problems
<<<OUTPUT ENDS>>>

(started) ['/bin/zsh', '-c', '''eslint'' -f unix --stdin --stdin-filename ''/Users/pgrove/git/ddi-guard-jira3/master/tests/active-directory.js'' < ''/var/folders/w_/nwg5jb2d5h19lhpv_rhx3c4c0000gn/T/nvimeHsmLZ/8/active-directory.js''']
(finished - exit code 1) ['/bin/zsh', '-c', '''eslint'' -f unix --stdin --stdin-filename ''/Users/pgrove/git/ddi-guard-jira3/master/tests/active-directory.js'' < ''/var/folders/w_/nwg5jb2d5h19lhpv_rhx3c4c0000gn/T/nvimeHsmLZ/9/active-directory.js''']

<<<OUTPUT STARTS>>>
/Users/pgrove/git/ddi-guard-jira3/master/tests/active-directory.js:2:1: Use the global form of 'use strict'. [Error/strict]
/Users/pgrove/git/ddi-guard-jira3/master/tests/active-directory.js:12:5: Expected object keys to be in ascending order. 'port' should be before 'target'. [Warning/sort-keys]
/Users/pgrove/git/ddi-guard-jira3/master/tests/active-directory.js:20:5: Expected object keys to be in ascending order. 'port' should be before 'target'. [Warning/sort-keys]
/Users/pgrove/git/ddi-guard-jira3/master/tests/active-directory.js:28:5: Expected object keys to be in ascending order. 'port' should be before 'target'. [Warning/sort-keys]
/Users/pgrove/git/ddi-guard-jira3/master/tests/active-directory.js:36:5: Expected object keys to be in ascending order. 'port' should be before 'target'. [Warning/sort-keys]
/Users/pgrove/git/ddi-guard-jira3/master/tests/active-directory.js:44:5: Expected object keys to be in ascending order. 'port' should be before 'target'. [Warning/sort-keys]
/Users/pgrove/git/ddi-guard-jira3/master/tests/active-directory.js:52:5: Expected object keys to be in ascending order. 'port' should be before 'target'. [Warning/sort-keys]
/Users/pgrove/git/ddi-guard-jira3/master/tests/active-directory.js:62:5: Expected object keys to be in ascending order. 'dnsDomainName' should be before 'username'. [Warning/sort-keys]
/Users/pgrove/git/ddi-guard-jira3/master/tests/active-directory.js:70:5: Expected object keys to be in ascending order. 'dnsDomainName' should be before 'username'. [Warning/sort-keys]
/Users/pgrove/git/ddi-guard-jira3/master/tests/active-directory.js:78:5: Expected object keys to be in ascending order. 'dnsDomainName' should be before 'username'. [Warning/sort-keys]
/Users/pgrove/git/ddi-guard-jira3/master/tests/active-directory.js:86:5: Expected object keys to be in ascending order. 'dnsDomainName' should be before 'username'. [Warning/sort-keys]

11 problems
<<<OUTPUT ENDS>>>

(finished - exit code 1) ['/bin/zsh', '-c', '''eslint'' -f unix --stdin --stdin-filename ''/Users/pgrove/git/ddi-guard-jira3/master/tests/active-directory.js'' < ''/var/folders/w_/nwg5jb2d5h19lhpv_rhx3c4c0000gn/T/nvimeHsmLZ/10/active-directory.js''']

<<<OUTPUT STARTS>>>
/Users/pgrove/git/ddi-guard-jira3/master/tests/active-directory.js:2:1: Use the global form of 'use strict'. [Error/strict]
/Users/pgrove/git/ddi-guard-jira3/master/tests/active-directory.js:12:5: Expected object keys to be in ascending order. 'port' should be before 'target'. [Warning/sort-keys]
/Users/pgrove/git/ddi-guard-jira3/master/tests/active-directory.js:20:5: Expected object keys to be in ascending order. 'port' should be before 'target'. [Warning/sort-keys]
/Users/pgrove/git/ddi-guard-jira3/master/tests/active-directory.js:28:5: Expected object keys to be in ascending order. 'port' should be before 'target'. [Warning/sort-keys]
/Users/pgrove/git/ddi-guard-jira3/master/tests/active-directory.js:36:5: Expected object keys to be in ascending order. 'port' should be before 'target'. [Warning/sort-keys]
/Users/pgrove/git/ddi-guard-jira3/master/tests/active-directory.js:44:5: Expected object keys to be in ascending order. 'port' should be before 'target'. [Warning/sort-keys]
/Users/pgrove/git/ddi-guard-jira3/master/tests/active-directory.js:52:5: Expected object keys to be in ascending order. 'port' should be before 'target'. [Warning/sort-keys]
/Users/pgrove/git/ddi-guard-jira3/master/tests/active-directory.js:62:5: Expected object keys to be in ascending order. 'dnsDomainName' should be before 'username'. [Warning/sort-keys]
/Users/pgrove/git/ddi-guard-jira3/master/tests/active-directory.js:70:5: Expected object keys to be in ascending order. 'dnsDomainName' should be before 'username'. [Warning/sort-keys]
/Users/pgrove/git/ddi-guard-jira3/master/tests/active-directory.js:78:5: Expected object keys to be in ascending order. 'dnsDomainName' should be before 'username'. [Warning/sort-keys]
/Users/pgrove/git/ddi-guard-jira3/master/tests/active-directory.js:86:5: Expected object keys to be in ascending order. 'dnsDomainName' should be before 'username'. [Warning/sort-keys]

11 problems
<<<OUTPUT ENDS>>>

(finished - exit code 1) ['/bin/zsh', '-c', '''eslint'' -f unix --stdin --stdin-filename ''/Users/pgrove/git/ddi-guard-jira3/master/tests/active-directory.js'' < ''/var/folders/w_/nwg5jb2d5h19lhpv_rhx3c4c0000gn/T/nvimeHsmLZ/11/active-directory.js''']

<<<OUTPUT STARTS>>>
/Users/pgrove/git/ddi-guard-jira3/master/tests/active-directory.js:2:1: Use the global form of 'use strict'. [Error/strict]
/Users/pgrove/git/ddi-guard-jira3/master/tests/active-directory.js:2:1: Unexpected var, use let or const instead. [Error/no-var]
/Users/pgrove/git/ddi-guard-jira3/master/tests/active-directory.js:4:1: Unexpected var, use let or const instead. [Error/no-var]
/Users/pgrove/git/ddi-guard-jira3/master/tests/active-directory.js:8:38: Unexpected function expression. [Error/prefer-arrow-callback]
/Users/pgrove/git/ddi-guard-jira3/master/tests/active-directory.js:9:22: Unexpected function expression. [Error/prefer-arrow-callback]
/Users/pgrove/git/ddi-guard-jira3/master/tests/active-directory.js:12:5: Expected object keys to be in ascending order. 'port' should be before 'target'. [Warning/sort-keys]
/Users/pgrove/git/ddi-guard-jira3/master/tests/active-directory.js:16:1: Trailing spaces not allowed. [Error/no-trailing-spaces]
/Users/pgrove/git/ddi-guard-jira3/master/tests/active-directory.js:17:27: Unexpected function expression. [Error/prefer-arrow-callback]
/Users/pgrove/git/ddi-guard-jira3/master/tests/active-directory.js:20:5: Expected object keys to be in ascending order. 'port' should be before 'target'. [Warning/sort-keys]
/Users/pgrove/git/ddi-guard-jira3/master/tests/active-directory.js:24:1: Trailing spaces not allowed. [Error/no-trailing-spaces]
/Users/pgrove/git/ddi-guard-jira3/master/tests/active-directory.js:25:26: Unexpected function expression. [Error/prefer-arrow-callback]
/Users/pgrove/git/ddi-guard-jira3/master/tests/active-directory.js:28:5: Expected object keys to be in ascending order. 'port' should be before 'target'. [Warning/sort-keys]
/Users/pgrove/git/ddi-guard-jira3/master/tests/active-directory.js:32:1: Trailing spaces not allowed. [Error/no-trailing-spaces]
/Users/pgrove/git/ddi-guard-jira3/master/tests/active-directory.js:33:31: Unexpected function expression. [Error/prefer-arrow-callback]
/Users/pgrove/git/ddi-guard-jira3/master/tests/active-directory.js:36:5: Expected object keys to be in ascending order. 'port' should be before 'target'. [Warning/sort-keys]
/Users/pgrove/git/ddi-guard-jira3/master/tests/active-directory.js:40:1: Trailing spaces not allowed. [Error/no-trailing-spaces]
/Users/pgrove/git/ddi-guard-jira3/master/tests/active-directory.js:41:28: Unexpected function expression. [Error/prefer-arrow-callback]
/Users/pgrove/git/ddi-guard-jira3/master/tests/active-directory.js:44:5: Expected object keys to be in ascending order. 'port' should be before 'target'. [Warning/sort-keys]
/Users/pgrove/git/ddi-guard-jira3/master/tests/active-directory.js:48:1: Trailing spaces not allowed. [Error/no-trailing-spaces]
/Users/pgrove/git/ddi-guard-jira3/master/tests/active-directory.js:49:32: Unexpected function expression. [Error/prefer-arrow-callback]
/Users/pgrove/git/ddi-guard-jira3/master/tests/active-directory.js:52:5: Expected object keys to be in ascending order. 'port' should be before 'target'. [Warning/sort-keys]
/Users/pgrove/git/ddi-guard-jira3/master/tests/active-directory.js:57:1: Trailing spaces not allowed. [Error/no-trailing-spaces]
/Users/pgrove/git/ddi-guard-jira3/master/tests/active-directory.js:58:28: Unexpected function expression. [Error/prefer-arrow-callback]
/Users/pgrove/git/ddi-guard-jira3/master/tests/active-directory.js:59:18: Unexpected function expression. [Error/prefer-arrow-callback]
/Users/pgrove/git/ddi-guard-jira3/master/tests/active-directory.js:62:5: Expected object keys to be in ascending order. 'dnsDomainName' should be before 'username'. [Warning/sort-keys]
/Users/pgrove/git/ddi-guard-jira3/master/tests/active-directory.js:66:1: Trailing spaces not allowed. [Error/no-trailing-spaces]
/Users/pgrove/git/ddi-guard-jira3/master/tests/active-directory.js:67:30: Unexpected function expression. [Error/prefer-arrow-callback]
/Users/pgrove/git/ddi-guard-jira3/master/tests/active-directory.js:70:5: Expected object keys to be in ascending order. 'dnsDomainName' should be before 'username'. [Warning/sort-keys]
/Users/pgrove/git/ddi-guard-jira3/master/tests/active-directory.js:74:1: Trailing spaces not allowed. [Error/no-trailing-spaces]
/Users/pgrove/git/ddi-guard-jira3/master/tests/active-directory.js:75:30: Unexpected function expression. [Error/prefer-arrow-callback]
/Users/pgrove/git/ddi-guard-jira3/master/tests/active-directory.js:78:5: Expected object keys to be in ascending order. 'dnsDomainName' should be before 'username'. [Warning/sort-keys]
/Users/pgrove/git/ddi-guard-jira3/master/tests/active-directory.js:82:1: Trailing spaces not allowed. [Error/no-trailing-spaces]
/Users/pgrove/git/ddi-guard-jira3/master/tests/active-directory.js:83:42: Unexpected function expression. [Error/prefer-arrow-callback]
/Users/pgrove/git/ddi-guard-jira3/master/tests/active-directory.js:86:5: Expected object keys to be in ascending order. 'dnsDomainName' should be before 'username'. [Warning/sort-keys]

34 problems
<<<OUTPUT ENDS>>>

(finished - exit code 1) ['/bin/zsh', '-c', '''eslint'' -f unix --stdin --stdin-filename ''/Users/pgrove/git/ddi-guard-jira3/master/tests/active-directory.js'' < ''/var/folders/w_/nwg5jb2d5h19lhpv_rhx3c4c0000gn/T/nvimeHsmLZ/12/active-directory.js''']

<<<OUTPUT STARTS>>>
/Users/pgrove/git/ddi-guard-jira3/master/tests/active-directory.js:2:1: Use the global form of 'use strict'. [Error/strict]
/Users/pgrove/git/ddi-guard-jira3/master/tests/active-directory.js:2:1: Unexpected var, use let or const instead. [Error/no-var]
/Users/pgrove/git/ddi-guard-jira3/master/tests/active-directory.js:4:1: Unexpected var, use let or const instead. [Error/no-var]
/Users/pgrove/git/ddi-guard-jira3/master/tests/active-directory.js:8:38: Unexpected function expression. [Error/prefer-arrow-callback]
/Users/pgrove/git/ddi-guard-jira3/master/tests/active-directory.js:9:22: Unexpected function expression. [Error/prefer-arrow-callback]
/Users/pgrove/git/ddi-guard-jira3/master/tests/active-directory.js:12:5: Expected object keys to be in ascending order. 'port' should be before 'target'. [Warning/sort-keys]
/Users/pgrove/git/ddi-guard-jira3/master/tests/active-directory.js:16:1: Trailing spaces not allowed. [Error/no-trailing-spaces]
/Users/pgrove/git/ddi-guard-jira3/master/tests/active-directory.js:17:27: Unexpected function expression. [Error/prefer-arrow-callback]
/Users/pgrove/git/ddi-guard-jira3/master/tests/active-directory.js:20:5: Expected object keys to be in ascending order. 'port' should be before 'target'. [Warning/sort-keys]
/Users/pgrove/git/ddi-guard-jira3/master/tests/active-directory.js:24:1: Trailing spaces not allowed. [Error/no-trailing-spaces]
/Users/pgrove/git/ddi-guard-jira3/master/tests/active-directory.js:25:26: Unexpected function expression. [Error/prefer-arrow-callback]
/Users/pgrove/git/ddi-guard-jira3/master/tests/active-directory.js:28:5: Expected object keys to be in ascending order. 'port' should be before 'target'. [Warning/sort-keys]
/Users/pgrove/git/ddi-guard-jira3/master/tests/active-directory.js:32:1: Trailing spaces not allowed. [Error/no-trailing-spaces]
/Users/pgrove/git/ddi-guard-jira3/master/tests/active-directory.js:33:31: Unexpected function expression. [Error/prefer-arrow-callback]
/Users/pgrove/git/ddi-guard-jira3/master/tests/active-directory.js:36:5: Expected object keys to be in ascending order. 'port' should be before 'target'. [Warning/sort-keys]
/Users/pgrove/git/ddi-guard-jira3/master/tests/active-directory.js:40:1: Trailing spaces not allowed. [Error/no-trailing-spaces]
/Users/pgrove/git/ddi-guard-jira3/master/tests/active-directory.js:41:28: Unexpected function expression. [Error/prefer-arrow-callback]
/Users/pgrove/git/ddi-guard-jira3/master/tests/active-directory.js:44:5: Expected object keys to be in ascending order. 'port' should be before 'target'. [Warning/sort-keys]
/Users/pgrove/git/ddi-guard-jira3/master/tests/active-directory.js:48:1: Trailing spaces not allowed. [Error/no-trailing-spaces]
/Users/pgrove/git/ddi-guard-jira3/master/tests/active-directory.js:49:32: Unexpected function expression. [Error/prefer-arrow-callback]
/Users/pgrove/git/ddi-guard-jira3/master/tests/active-directory.js:52:5: Expected object keys to be in ascending order. 'port' should be before 'target'. [Warning/sort-keys]
/Users/pgrove/git/ddi-guard-jira3/master/tests/active-directory.js:57:1: Trailing spaces not allowed. [Error/no-trailing-spaces]
/Users/pgrove/git/ddi-guard-jira3/master/tests/active-directory.js:58:28: Unexpected function expression. [Error/prefer-arrow-callback]
/Users/pgrove/git/ddi-guard-jira3/master/tests/active-directory.js:59:18: Unexpected function expression. [Error/prefer-arrow-callback]
/Users/pgrove/git/ddi-guard-jira3/master/tests/active-directory.js:62:5: Expected object keys to be in ascending order. 'dnsDomainName' should be before 'username'. [Warning/sort-keys]
/Users/pgrove/git/ddi-guard-jira3/master/tests/active-directory.js:66:1: Trailing spaces not allowed. [Error/no-trailing-spaces]
/Users/pgrove/git/ddi-guard-jira3/master/tests/active-directory.js:67:30: Unexpected function expression. [Error/prefer-arrow-callback]
/Users/pgrove/git/ddi-guard-jira3/master/tests/active-directory.js:70:5: Expected object keys to be in ascending order. 'dnsDomainName' should be before 'username'. [Warning/sort-keys]
/Users/pgrove/git/ddi-guard-jira3/master/tests/active-directory.js:74:1: Trailing spaces not allowed. [Error/no-trailing-spaces]
/Users/pgrove/git/ddi-guard-jira3/master/tests/active-directory.js:75:30: Unexpected function expression. [Error/prefer-arrow-callback]
/Users/pgrove/git/ddi-guard-jira3/master/tests/active-directory.js:78:5: Expected object keys to be in ascending order. 'dnsDomainName' should be before 'username'. [Warning/sort-keys]
/Users/pgrove/git/ddi-guard-jira3/master/tests/active-directory.js:82:1: Trailing spaces not allowed. [Error/no-trailing-spaces]
/Users/pgrove/git/ddi-guard-jira3/master/tests/active-directory.js:83:42: Unexpected function expression. [Error/prefer-arrow-callback]
/Users/pgrove/git/ddi-guard-jira3/master/tests/active-directory.js:86:5: Expected object keys to be in ascending order. 'dnsDomainName' should be before 'username'. [Warning/sort-keys]

34 problems
<<<OUTPUT ENDS>>>
```

I should note that running run this:
```bash
> /bin/zsh -c "'eslint' --config '/Users/pgrove/git/ddi-guard-jira3/master/tests/.eslintrc.js' --fix '/Users/pgrove/git/ddi-guard-jira3/master/tests/json_api.js'"

/Users/pgrove/git/ddi-guard-jira3/master/tests/json_api.js
    1:1   error    Use the global form of 'use strict'                                    strict
    1:5   warning  'should' is never reassigned. Use 'const' instead                      prefer-const
    2:2   warning  'JsonApi' is never reassigned. Use 'const' instead                     prefer-const
    3:2   warning  'spawn' is never reassigned. Use 'const' instead                       prefer-const
    4:2   warning  'Promise' is never reassigned. Use 'const' instead                     prefer-const
    5:1   error    '_' is not defined                                                     no-undef
    8:7   error    'servers' is assigned a value but never used                           no-unused-vars
   10:5   error    Identifier 'server_path' is not in camel case                          camelcase
   10:22  error    Use path.join() or path.resolve() instead of + to create paths         no-path-concat
   13:2   error    Mixed spaces and tabs                                                  no-mixed-spaces-and-tabs
   15:1   error    Identifier 'server_path' is not in camel case                          camelcase
   15:15  error    Identifier 'server_path' is not in camel case                          camelcase
   41:29  error    Unexpected mix of '+' and '*'                                          no-mixed-operators
   41:33  error    Unexpected mix of '+' and '*'                                          no-mixed-operators
   43:5   warning  Expected an assignment or function call and instead saw an expression  no-unused-expressions
   44:5   warning  Expected an assignment or function call and instead saw an expression  no-unused-expressions
   56:5   warning  'finish' is never reassigned. Use 'const' instead                      prefer-const
   60:23  error    '_' is not defined                                                     no-undef
  100:4   error    Unexpected var, use let or const instead                               no-var
  119:27  warning  Expected error to be handled                                           handle-callback-err
  140:10  error    Identifier 'check_closed' is not in camel case                         camelcase
  149:35  error    'error' is already declared in the upper scope                         no-shadow
  152:36  error    'error' is already declared in the upper scope                         no-shadow
  166:10  error    Identifier 'expect_count' is not in camel case                         camelcase
  167:12  warning  Expected error to be handled                                           handle-callback-err
  167:26  error    'res' is defined but never used                                        no-unused-vars
  170:15  error    Expected '===' and instead saw '=='                                    eqeqeq
  183:5   warning  Expected an assignment or function call and instead saw an expression  no-unused-expressions
  185:6   warning  Expected an assignment or function call and instead saw an expression  no-unused-expressions
  195:28  warning  Expected error to be handled                                           handle-callback-err
  201:5   warning  Expected an assignment or function call and instead saw an expression  no-unused-expressions

✖ 31 problems (18 errors, 13 warnings)

```

The file does get fixed, and I see exit 0, is the temporary file not replacing the working buffer after completion somehow?

--- a/autoload/ale/fixers/eslint.vim
+++ b/autoload/ale/fixers/eslint.vim
@@ -11,7 +11,7 @@ function! ale#fixers#eslint#Fix(buffer) abort
 
     return {
     \   'command': ale#node#Executable(a:buffer, l:executable)
-    \       . ' --config ' . ale#Escape(l:config)
+    \       . ' -c ' . ale#Escape(l:config)
     \       . ' --fix %t',
     \   'read_temporary_file': 1,
     \}

--- a/test/fixers/test_eslint_fixer_callback.vader
+++ b/test/fixers/test_eslint_fixer_callback.vader
@@ -13,7 +13,7 @@ Execute(The path to eslint.js should be run on Unix):
   \   'read_temporary_file': 1,
   \   'command':
   \     ale#Escape(simplify(g:dir . '/../eslint-test-files/react-app/node_modules/eslint/bin/eslint.js'))
-  \     . ' --config ' . ale#Escape(simplify(g:dir . '/../eslint-test-files/react-app/.eslintrc.js'))
+  \     . ' -c ' . ale#Escape(simplify(g:dir . '/../eslint-test-files/react-app/.eslintrc.js'))
   \     . ' --fix %t',
   \ },
   \ ale#fixers#eslint#Fix(bufnr(''))
@@ -26,7 +26,7 @@ Execute(The lower priority configuration file in a nested directory should be pr
   \   'read_temporary_file': 1,
   \   'command':
   \     ale#Escape(simplify(g:dir . '/../eslint-test-files/react-app/node_modules/eslint/bin/eslint.js'))
-  \     . ' --config ' . ale#Escape(simplify(g:dir . '/../eslint-test-files/react-app/subdir-with-config/.eslintrc'))
+  \     . ' -c ' . ale#Escape(simplify(g:dir . '/../eslint-test-files/react-app/subdir-with-config/.eslintrc'))
   \     . ' --fix %t',
   \ },
   \ ale#fixers#eslint#Fix(bufnr(''))
@@ -39,7 +39,7 @@ Execute(package.json should be used as a last resort):
   \   'read_temporary_file': 1,
   \   'command':
   \     ale#Escape(simplify(g:dir . '/../eslint-test-files/react-app/node_modules/eslint/bin/eslint.js'))
-  \     . ' --config ' . ale#Escape(simplify(g:dir . '/../eslint-test-files/react-app/.eslintrc.js'))
+  \     . ' -c ' . ale#Escape(simplify(g:dir . '/../eslint-test-files/react-app/.eslintrc.js'))
   \     . ' --fix %t',
   \ },
   \ ale#fixers#eslint#Fix(bufnr(''))
@@ -51,7 +51,7 @@ Execute(package.json should be used as a last resort):
   \   'read_temporary_file': 1,
   \   'command':
   \     ale#Escape(simplify(g:dir . '/../eslint-test-files/node_modules/.bin/eslint'))
-  \     . ' --config ' . ale#Escape(simplify(g:dir . '/../eslint-test-files/package.json'))
+  \     . ' -c ' . ale#Escape(simplify(g:dir . '/../eslint-test-files/package.json'))
   \     . ' --fix %t',
   \ },
   \ ale#fixers#eslint#Fix(bufnr(''))

